client/lib/analytics/ga.js

```diff
@@ -18,6 +19,19 @@ function initialize() {
 			...getGoogleAnalyticsDefaultConfig(),
 		};
 
+		// We enable custom cross-domain linking only for Akismet checkouts
+		if ( isAkismetCheckout() ) {
+			const queryParams = new URLSearchParams( location.search );
+			const gl = queryParams.get( '_gl' );
+
+			// If we have a _gl query param, cross-domain linking is done automatically
+			if ( ! gl ) {
+				// Setting cross-domain manually: https://support.google.com/analytics/answer/10071811?hl=en#zippy=%2Cmanual-setup
+				params.client_id = queryParams.get( '_gl_cid' );
```

Comment:
Do we need to do any sort of validation or sanitization when pulling these query parameters here? It's possible that the gtag script handles that lower down

Reply:
From what I know, this is what `URLSearchParams` interface is responsible for. `queryParams.get()` is parsing values already as specified here (https://url.spec.whatwg.org/#concept-urlencoded-parser). Additionally, values used as `_gl_cid` and `_gl_sid` by design are not containing any characters that need to be sanitized.